CMakeLists.txt

```diff
@@ -534,12 +542,12 @@ if (srt_libspec_shared)
 	set_target_properties (${TARGET_srt}_shared PROPERTIES VERSION ${SRT_VERSION} SOVERSION ${SRT_VERSION_MAJOR})
 	list (APPEND INSTALL_TARGETS ${TARGET_srt}_shared)
 	if (ENABLE_ENCRYPTION)
-		target_link_libraries(${TARGET_srt}_shared PRIVATE ${SSL_LIBRARIES})
+		target_link_libraries(${TARGET_srt}_shared PRIVATE ${SSL_LIBRARIES} ${GNUSTL_LIBRARIES} ${GNUSTL_LDFLAGS})
```

Comment:
From cmake's doc for `target_link_libraries`: *Repeated calls for the same <target> append items in the order called.*

https://cmake.org/cmake/help/v3.3/command/target_link_libraries.html

Why don't you make an additional call for this, under a condition with `USE_GNUSTL` and selected systems.